### PR TITLE
非同期通信

### PR DESCRIPTION
**Pull request by yoshibuchi**

### CODE REVIEW

File: app/assets/javascripts/message.js [hunk 1]
```diff
@@ -1,33 +1,50 @@
...
+        ${message.content}
+        ${image}
+        
+      </div>
+    </div>`
       return html;
     }
-    $('#new_comment').on('submit', function(e){
+    $("#new_message").on('submit', function(e){
+      // console.log(e)
```
Comment (TOMARUMARU) [inline app/assets/javascripts/message.js]: 不要なコメントアウトは消しましょう

File: app/assets/javascripts/message.js [hunk 2]
```diff
@@ -1,33 +1,50 @@
...
       e.preventDefault();
       var formData = new FormData(this);
-      var href = window.location.href + '/comments'
+      var text = $(".form__message").val()
+      var url = $(this).attr('action');
+      console.log(text)
+      console.log(url)
+      console.log('go')
```
Comment (TOMARUMARU) [inline app/assets/javascripts/message.js]: デバッグ消しましょう

File: app/assets/javascripts/message.js [hunk 3]
```diff
@@ -1,33 +1,50 @@
...
-        url: href,
+        url: url,
         type: "POST",
         data: formData,
         dataType: 'json',
         processData: false,
         contentType: false
       })
       .done(function(data){
+        console.log(data)
```
Comment (TOMARUMARU) [inline app/assets/javascripts/message.js]: こちらも

File: app/assets/javascripts/message.js [hunk 4]
```diff
@@ -1,33 +1,50 @@
...
         var html = buildHTML(data);
-        $('.comments').append(html)
-        $('.textbox').val('')
-      .fail(function(){
-        alert('error');
-        }) 
+        $('.form__submit-button').prop('disabled', false);
+        $('.messages').append(html);
+        $('.textbox').val('');
```
Comment (TOMARUMARU) [inline app/assets/javascripts/message.js]: resetメソッドを使いましょう

File: app/views/messages/create.json.jbuilder [hunk 5]
```diff
@@ -0,0 +1,7 @@
+json.content  @message.content
+json.user_id  @message.user.id
```
Comment (TOMARUMARU) [inline app/views/messages/create.json.jbuilder]: user_idは使用しないですかね？

File: app/views/messages/create.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+json.user_name  @message.user.name
+json.id @message.id
+json.date @message.created_at.strftime("%Y/%m/%d %H:%M")
+json.group_id @message.group_id
```
Comment (TOMARUMARU) [inline app/views/messages/create.json.jbuilder]: group_idも

File: app/controllers/comments_controller.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+def create
+    @comment = Comment.create(text: comment_params[:text], group_id: comment_params[:group_id], user_id: current_user.id)
```
Comment (kuroko918) [inline app/controllers/comments_controller.rb]: use_idはcomment_params内でmergeしましょう！

Comment (kuroko918) [reply app/controllers/comments_controller.rb]: そうすると、create(comment_params)でいけるかと

File: app/controllers/messages_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    context 'not log in' do
+      before do
+        get :index, params: { group_id: group.id }
+      end
+
+      it 'redirects to new_user_session_path' do
+        expect(response).to redirect_to(new_user_session_path)
+      end
+
+      describe '#create' do
```
Comment (kuroko918) [inline app/controllers/messages_controller_spec.rb]: みにくいので、describe “#index”の中にネストさせないようにしましょう。

File: app/views/layouts/application.html.haml [hunk 9]
```diff
@@ -6,6 +6,7 @@
     = stylesheet_link_tag    'application', media: 'all', data: { turbolinks: { track: true }}
     = javascript_include_tag 'application', data: { turbolinks: { track: true }}
     = csrf_meta_tags
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
```
Comment (kuroko918) [inline app/views/layouts/application.html.haml]: CDNでなく、gemで対応しましょう

File: app/controllers/messages_controller.rb [hunk 10]
```diff
@@ -8,7 +8,10 @@ def index
     def create
       @message = @group.messages.new(message_params)
       if @message.save
-        redirect_to group_messages_path(@group), notice: 'メッセージが送信されました'
+        respond_to do |format|
+          format.html { redirect_to "group_messages_path(params[:group_id]", notice: 'メッセージが送信されました'  }
```
Comment (kuroko918) [inline app/controllers/messages_controller.rb]: format.htmlは使用しないので、全部消すとエラーが出ることがあるので、format.html以外、この行は消してください

File: app/controllers/messages_controller_spec.rb (hunk 8 above)
Comment (yoshibuchi) [reply app/controllers/messages_controller_spec.rb]: 非同期のプルリクエストのみレビューお願いします。
すいません、間違えました。

File: Gemfile [hunk 11]
```diff
@@ -56,4 +60,5 @@ gem 'erb2haml'
 gem 'font-awesome-rails'
 gem 'devise'
 gem 'carrierwave'
-gem 'mini_magick'
\ No newline at end of file
+gem 'mini_magick'
+gem 'pry-rails'
```
Comment (kuroko918) [inline Gemfile]: productionでは使用しないので、test, developmentのgroupの中に入れて下さい！

File: app/assets/stylesheets/modules/_chat_main.scss [hunk 12]
```diff
@@ -92,7 +92,7 @@
         padding: 0 0 0 10px;
       }
       &__input-box {
-        padding: 0 40px 0 10px;
+        padding: 0 0 0 10px;
```
Comment (kuroko918) [inline app/assets/stylesheets/modules/_chat_main.scss]: padding-leftでいいかと！

File: app/controllers/messages_controller_spec.rb (hunk 8 above)
Comment (kuroko918) [reply app/controllers/messages_controller_spec.rb]: このブランチに必要でないのでしたら、pushしない様にしましょう！

**Comment by kuroko918** [comment]
https://github.com/yoshibuchi/chat-space/pull/3/files#r294248558
こちら如何ですか？

**Comment by kuroko918** [comment]
変更が反映されていないみたいです！
